gen/src/write.rs

```diff
@@ -373,6 +381,34 @@ fn write_enum(out: &mut OutFile, enm: &Enum) {
     writeln!(out, "}};");
 }
 
+fn check_enum(out: &mut OutFile, enm: &Enum) {
+    let discriminants = enm
+        .variants
+        .iter()
+        .scan(None, |prev_discriminant, variant| {
+            let discriminant = variant
+                .discriminant
+                .unwrap_or_else(|| prev_discriminant.map_or(0, |n| n + 1));
+            *prev_discriminant = Some(discriminant);
+            Some(discriminant)
+        });
+    writeln!(
+        out,
+        "static_assert(sizeof({}) == sizeof(uint32_t));",
+        enm.ident
+    );
+    enm.variants
+        .iter()
+        .zip(discriminants)
+        .for_each(|(variant, discriminant)| {
+            writeln!(
+                out,
+                "static_assert({}::{} == {});",
```

Comment:
Let's add a message argument here to support pre-C++17.
https://en.cppreference.com/w/cpp/language/static_assert
Something like `"disagrees with the value in #[cxx::bridge]"`.

Reply:
Separately, it looks like this expands to an expression like `E::A == 0`? That didn't work on my machine.

```console
$ cat main.cc 
#include <cstdint>

enum class E : uint32_t {
  A = 0,
};

static_assert(E::A == 0);
```

```console
$ clang++ main.cc -std=c++17
main.cc:7:20: error: invalid operands to binary expression ('E' and 'int')
static_assert(E::A == 0);
              ~~~~ ^  ~
1 error generated.
```

```console
$ g++ main.cc -std=c++17
main.cc:7:20: error: no match for ‘operator==’ (operand types are ‘E’ and ‘int’)
    7 | static_assert(E::A == 0);
      |               ~~~~ ^~ ~
      |                  |    |
      |                  E    int
```

Reply:
Thanks for the C++17 tip.  I didn't know that.

About the miscompile, in my test code I was just doing "enum A" and not "enum class A", in which case it compiled.  But I added static_casts so it works both ways.  Thanks!